chainercv/links/model/faster_rcnn/utils/proposal_target_creator.py

```diff
@@ -0,0 +1,136 @@
+    """Assign ground truth bounding boxes to given RoIs.
+
+    The :meth:`__call__` of this class generates training targets
+    for each object proposal.
+    This is used to train Faster RCNN [#]_.
+
+    .. [#] Shaoqing Ren, Kaiming He, Ross Girshick, Jian Sun. \
+    Faster R-CNN: Towards Real-Time Object Detection with \
+    Region Proposal Networks. NIPS 2015.
+
+    Args:
+        n_sample (int): The number of sampled regions.
+        fg_fraction (float): Fraction of regions that is labeled as a
+            foreground.
+        fg_thresh (float): IoU threshold for a RoI to be considered as a
+            foreground.
+        bg_thresh_hi (float): RoI is considered to be the background if IoU
+            is in [:obj:`bg_thresh_hi`, :obj:`bg_thresh_hi`).
+        bg_thresh_lo (float): See above.
+
+    """
+
+    def __init__(self,
+                 n_sample=128,
+                 fg_fraction=0.25,
+                 fg_thresh=0.5, bg_thresh_hi=0.5, bg_thresh_lo=0.0
+                 ):
+        self.n_sample = n_sample
+        self.fg_fraction = fg_fraction
+        self.fg_thresh = fg_thresh
+        self.bg_thresh_hi = bg_thresh_hi
+        self.bg_thresh_lo = bg_thresh_lo
+
+    def __call__(self, roi, bbox, label,
+                 loc_normalize_mean=(0., 0., 0., 0.),
+                 loc_normalize_std=(0.1, 0.1, 0.2, 0.2)):
+        """Assigns ground truth to sampled proposals.
+
+        This function samples total of :obj:`self.n_sample` RoIs
+        from the combination of :obj:`roi` and :obj:`bbox`.
+        The RoIs are assigned with the ground truth class labels as well as
+        bounding box offsets and scales to match the ground truth bounding
+        boxes. As many as :obj:`fg_fraction * self.n_sample` RoIs are
+        sampled as foregrounds.
+
+        Offsets and scales of bounding boxes are calculated using
+        :func:`chainercv.links.model.faster_rcnn.bbox2loc`.
+        Also, types of input arrays and output arrays are same.
+
+        Here are notations.
+
+        * :math:`S` is the total number of sampled RoIs, which equals \
+            :obj:`self.n_sample`.
+        * :math:`L` is number of object classes possibly including the \
+            background.
+
+        Args:
+            roi (array): Region of interests from which we sample.
+                Its shape is :math:`(R, 4)`
+            bbox (array): The coordinates of ground truth bounding boxes.
+                Its shape is :math:`(R', 4)`.
+            label (array): Ground truth bounding box labels. Its shape
+                is :math:`(R',)`.
+            loc_normalize_mean (tuple of four floats): Mean values to normalize
+                coordinates of bouding boxes.
+            loc_normalize_std (tupler of four floats): Standard deviation of
+                the coordinates of bounding boxes.
+
+        Returns:
+            (array, array, array):
+
+            * **sample_roi**: Regions of interests that are sampled. \
```

Comment:
For the consistency with `IoU`, `Regions of interests` -> `Regions of Interests`?
How about `Sampled Regions of Interests (RoIs).`?